greenplumpython/group.py

```diff
@@ -20,22 +20,33 @@ def __init__(self, table: "Table", grouping_sets: List[Iterable["Expr"]]) -> Non
         self._table = table
         self._grouping_sets = grouping_sets
 
-    def apply(self, func: Callable[["Table"], "FunctionExpr"]) -> "FunctionExpr":
+    def assign(self, **new_columns: Callable[["Table"], Any]) -> "Table":
         """
-        Apply a function to the grouping set.
+        Assigns new columns to the current grouping sets. Existing columns
+        cannot be reassigned.
 
         Args:
-            func: An aggregate function to be applied to
+            new_columns:
 
         Returns:
-            FunctionExpr: a callable
-
-        Example:
-            .. code-block::  python
-
-                numbers.group_by("is_even").apply(lambda row: count(row["*"]))
+            :class:`Table` with the new columns.
         """
-        return func(self._table).bind(group_by=self)
+        from greenplumpython.table import Table
+
+        if len(new_columns) == 0:
+            return self
+        targets: List[str] = []
+        if len(new_columns):
```

Comment:
we if len(new_columns) in the first and return early there is no need if len(new_columns) IMO.